empirical-results.tex

```diff
@@ -1 +1,29 @@
-\section{Empirical Results}
\ No newline at end of file
+\section{Empirical Results}
+
+\newcommand{\rref}[1]{}
+
+In order to evaluate our consumption mechanisms, we implement the federated consumption mechanism in Solidity. We provide a concrete implementation of the \textsf{burn-verifier} contract described in \rref{alg.burn-verifier} as \textsf{BurnedBTC}. We implement the \textsf{Crosschain} parent contract from \cite{pow-sidechains}. In order to achieve this we implement our own verification methods similar to \rref{alg.verify-event-federation} in our \textsf{Verifier} library. Specifically, we verify transaction data in \textsf{Verifier} similarly to \rref{alg.verify-tx} by making use of the open-source bitcoin-spv library \cite{bitcoin-spv-library}. We verify block connection using a Merkle Mountain Range where the leaf corresponding to each block has the value of the block's transaction id merkle tree root.
+
+Finally, the \textsf{BurnVerifier} library provides a concrete implementation for our berning protocol. Specifically the library implements the \textsf{BurnVerify} function of the protocol.
+
+In our implementation, the federation is limited to one party for simplicity but it can be easily adapted for more.
+
+We release our implementation as open-source software under the MIT license
+\footnote{\url{https://github.com/decrypto-org/burn-paper/tree/master/experiment}}.
+The implementation is fully tested with 100\% code coverage.
+
+We assume a median gas price of 6.9 gwei, which was gathered at the time of writing. We also assume an Ethereum price of \$170.07. This gives us a gas price of $\$1.173483 * 10^{-6}$.
+
+\begin{center}
+    \begin{tabular}{ |c|c|c| } 
+     \hline
+     Method & Gas cost & Equivalent in USD \\
+     \hline
+     \textsf{CheckpointRepo.vote} & 43039 gas & $\$0.05$ \\
+     \textsf{BurnedBTC.claim} & 232926 gas & $\$0.27$ \\
+     \hline
+    \end{tabular}
+\end{center}
+
+For the end user to claim his burn, the cost is thus \$0.27. Comparatively, for a Bitcoin transaction to be included in the next block at the time of writing a user would have to spend \$0.77
+\footnote{\url{http://archivecaslytosk.onion/2019.08.30-142942/https://bitcoinfees.info/}}.
```

Comment:
What is this address?

Reply:
explained IRL, would you prefer this removed?

Reply:
could you remind me what the "Accessed at" footnote should look like exactly? Or should it be a citation?